src/route/api.ts

```diff
@@ -1,23 +1,87 @@
 import { Request, Response, Router } from 'express';
 import multer from 'multer';
+import pino from 'pino';
 
-import { processCSV } from '../controllers/csv-processor';
+import { processCSV, uploadCSV } from '../controllers/csv-processor';
+import { listFilesInDirectory, downloadFileFromDataLake } from '../controllers/datalake';
 
 const storage = multer.memoryStorage();
 const upload = multer({ storage });
 
+export const logger = pino({
+    name: 'StatsWales-Alpha-App',
+    level: 'debug'
+});
+
 export const apiRoute = Router();
 
+logger.debug('API is runnuing...');
+
 apiRoute.get('/', (req, res) => {
     res.json({ message: 'API is available' });
 });
 
 apiRoute.post('/csv', upload.single('csv'), async (req: Request, res: Response) => {
+    // const page_number_str: string = req.query.page_number || req.body?.page_number;
+    // const page_size_str: string = req.query.page_size || req.body?.page_size;
+    // const page_number: number = Number.parseInt(page_number_str, 10) || 1;
+    // const page_size: number = Number.parseInt(page_size_str, 10) || 100;
+    const filemame: string | undefined = req.query.filename?.toString() || req.file?.originalname;
+    if (filemame === undefined) {
+        res.status(400);
+        res.json({
+            success: false,
+            headers: undefined,
+            data: undefined,
+            errors: [
+                {
+                    field: 'filename',
+                    message: 'No filename provided'
+                }
+            ]
+        });
+        return;
+    }
+    const processedCSV = await uploadCSV(req.file?.buffer, filemame);
```

Comment:
filename is a user supplied value and therefore untrusted.
We cannot store the file in the datalake with this name because
1a) it might allow a path injection
1b) ...and therefore allow the user to overwrite a specific file of their choice.
2) 2 files with the same name might legitimately clash and we would not be able to retain them both.

We also need to keep track of the mime-type that the user sends us (not that we can reliably trust that either, but it gives us a starting point for validation).

Therefore, we should consider storing the file metadata (original file name / user filename, size, mime-type and where we put it in the datalake) in some metadata storage somewhere.
Then, when as we stream the file from the client we can generate a filename for it based on its contents (sha-256 perhaps?) and then assign that name in the datalake.